node/grpc/server_v2.go

```diff
@@ -367,3 +380,62 @@ func (s *ServerV2) GetChunks(ctx context.Context, in *pb.GetChunksRequest) (*pb.
 		ChunkEncodingFormat: pb.ChunkEncodingFormat_GNARK,
 	}, nil
 }
+
+// validateDispersalRequest validates the DisperseBlobRequest and returns the blob header
+// Differences between this and the DispersalServerV2 are:
+// - Takes *corev2.BlobCertificate instead of DisperseBlobRequest
+// - no encoding prover GetCommitmentsForPaddedLength check
+// - directly take blob lengths (no blob data yet)
+// - doesn't check every 32 bytes is a valid field element
+func (s *ServerV2) validateDispersalRequest(
```

Comment:
Interesting we didn't validate the blob certificates from the request before

Reply:
yeah I think we just trusted the disperser for those checks:p